registry/remote/repository.go

```diff
@@ -102,6 +102,9 @@ type Repository struct {
 	// If less than or equal to zero, a default (currently 4MiB) is used.
 	MaxMetadataBytes int64
 
+	// ReferrersGC specifies whether to delete the dangling referrers index.
+	ReferrersGC bool
```

Comment:
1. Wondering if we should call it `SkipReferrersGC` since we perform GC by default🤔. But if we plan to skip GC by default **in the future**, `ReferrersGC` might be a more appropriate name?
2. Whatever option name and default value we opt to, we should explicitly call out the behavior and the default value in the comment documentation.
3. If the default value is `true`, we should make sure it's correctly set in other places like [NewRegistry](https://github.com/oras-project/oras-go/blob/v2.2.0/registry/remote/registry.go#L53-L65) and [newRepositoryWithOptions](https://github.com/oras-project/oras-go/blob/v2.2.0/registry/remote/repository.go#L140-L153). (See the comment at line108😀)

Reply:
>  But if we plan to skip GC by default in the future, ReferrersGC might be a more appropriate name?

Skip GC by default is breaking to oras-go and can be done on CLI side. Since oras CLI is the only user of it and don't require the default change, it's better not to introduce this change.

Reply:
Yeah we shouldn't introduce any breaking change in this PR. I just wanted to discuss the name.

Reply:
`SkipReferrersGC` (with default value false) and `ReferrersGC` (with default value true) both make sense to me, cc @FeynmanZhou @yizha1 for more advice on the naming.